uaa_client/authentication.py

```diff
@@ -93,13 +93,21 @@ def get_user_by_email(email):
         of security policies for logins.
         '''
 
+        APPROVED_DOMAINS = getattr(settings, 'UAA_APPROVED_DOMAINS', [])
+
         try:
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
-            logger.info(
-                'User with email {} does not exist'.format(email)
-            )
-            return None
+            email_pieces = email.split('@')
+            if email_pieces[1] in APPROVED_DOMAINS:
+                User.objects.create_user(email_pieces[0], email)
```

Comment:
Also, I might be getting a bit too architecture-astronauty here, but I wonder if it'd make sense to factor out line 103 into its own method called `create_user_for_email()`.  This would make it super easy for projects to subclass the backend and provide e.g. slightly different username-naming schemes, or add some default permissions to the created user based on their domain, or something else entirely, because they'd just have to override the one `create_user_for_email()` method.